src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,231 @@
+			timeout: REQUEST_TIMEOUT
+		};
+		return request(oauthoptions).then(result => {
+			if (result) {
+				const parsedResult = JSON.parse(result);
+				return parsedResult.access_token;
+			}
+			// eslint-disable-next-line no-console
+			console.error('Oauth failed');
+			return null;
+		});
+	}
+
+	checkEnv() {
+		return (
+			DOMAIN &&
+			USER &&
+			PASSWORD &&
+			GRANT_TYPE &&
+			OAUTH_SECRET &&
+			CLIENT_ID
+		);
+	}
+
+	async login() {
+		this.authorization = await this.getCookie();
+		this.accessToken = await this.getAuth();
+	}
+
+	isLoggedin() {
+		// returns false if cookie or accesstoken is falsy
+		return !!this.authorization && !!this.accessToken;
+	}
+
+	async GET(data) {
+		// run twice, one with
+		const contentType = data.query.contentType || 'FILES'; // enum:[FILES,FILES_AND_FOLDERS,COLLECTIONS,ALL]
+		const skipCount = data.query.$skip || 0;
+		const maxItems = data.query.$limit || 9;
+		const sortProperties = data.query.sortProperties || 'score';
+		const sortAscending = data.query.$ascending || true;
+		const propertyFilter = data.query.propertyFilter || '-all-'; // '-all-' for all properties OR ccm-stuff
+		const searchWord = data.query.searchQuery || 'img'; // will give pictures of flowers as default
+
+		// const filterOptions = data.query.filterOptions
+
+
+		if (!this.checkEnv()) {
+			return 'Update your env variables. See --> src/services/edusharing/envTemplate';
+		}
+
+		if (this.isLoggedin() === false) {
+			await this.login();
+		}
+		const options = {
+			method: 'POST',
+			// This will be changed later with a qs where sorting, filtering etc is present.
+			// eslint-disable-next-line max-len
+			url: `${DOMAIN}/edu-sharing/rest/search/v1/queriesV2/mv-repo.schul-cloud.org/mds/ngsearch/?contentType=${contentType}&skipCount=${skipCount}&maxItems=${maxItems}&sortProperties=${sortProperties}&sortProperties=cm%3Amodified&sortAscending=${sortAscending}&sortAscending=false&propertyFilter=${propertyFilter}&`,
+			headers: {
+				...EduSharingConnector.headers,
+				cookie: this.authorization,
+			},
+			body: JSON.stringify({
+				criterias: [
+					{ property: 'ngsearchword', values: [`${searchWord}`] }
+				],
+				facettes: ['cclom:general_keyword']
+			}),
+			timeout: REQUEST_TIMEOUT,
+		};
+
+		// code below includes params in seperate key/value, but does not work.
+
+
+
+		const eduResponse = await request(options);
+		let parsed = JSON.parse(eduResponse);
+
+		// provided by client eg data.query.filterOptions
+		const filterOptions = {
+			mimetype: ['text/html', 'image/jpeg'],
+			provider: ['BauhausMaterial.de', 'München educationcenter', 'Khan Academy']
+		}
+
+		// checks if user has set filter options
+		if (!!Object.values(filterOptions).length) {
+			parsed = filterResult(parsed, filterOptions)
+		}
+
+		function filterResult(obj, options) {
+			let result;
+			// checks if user has set type filter
+			if (options.mimetype.length) {
+				result = filterMime(obj, options.mimetype)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['filterMime' was used before it was defined.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4991925)